internal/metamorph/processor_helpers.go

```diff
@@ -22,64 +23,95 @@ func (p *Processor) GetProcessorMapSize() int {
 	return p.responseProcessor.getMapLen()
 }
 
-func updateStatusMap(statusUpdatesMap map[chainhash.Hash]store.UpdateStatus, statusUpdate store.UpdateStatus) {
-	foundStatusUpdate, found := statusUpdatesMap[statusUpdate.Hash]
+func (p *Processor) updateStatusMap(statusUpdate store.UpdateStatus) error {
+	currentStatusUpdate, err := p.getTransactionStatus(statusUpdate.Hash)
+	if err != nil {
+		if errors.Is(err, cache.ErrCacheNotFound) {
+			// if record doesn't exist, save new one
+			return p.setTransactionStatus(statusUpdate)
+		}
+		return err
+	}
 
-	if !found || shouldUpdateStatus(statusUpdate, foundStatusUpdate) {
+	if shouldUpdateStatus(statusUpdate, *currentStatusUpdate) {
 		if len(statusUpdate.CompetingTxs) > 0 {
-			statusUpdate.CompetingTxs = mergeUnique(statusUpdate.CompetingTxs, foundStatusUpdate.CompetingTxs)
+			statusUpdate.CompetingTxs = mergeUnique(statusUpdate.CompetingTxs, currentStatusUpdate.CompetingTxs)
 		}
-
-		statusUpdatesMap[statusUpdate.Hash] = statusUpdate
+		// TODO: combine status history
 	}
-}
 
-func (p *Processor) updateStatusMap(statusUpdate store.UpdateStatus) (map[chainhash.Hash]store.UpdateStatus, error) { //nolint:unused
-	statusUpdatesMap := p.getStatusUpdateMap()
+	return p.setTransactionStatus(statusUpdate)
+}
 
-	foundStatusUpdate, found := statusUpdatesMap[statusUpdate.Hash]
+func (p *Processor) setTransactionStatus(status store.UpdateStatus) error {
+	bytes, err := json.Marshal(status)
+	if err != nil {
+		return errors.Join(ErrFailedToSerialize, err)
+	}
 
-	if !found || shouldUpdateStatus(statusUpdate, foundStatusUpdate) {
-		if len(statusUpdate.CompetingTxs) > 0 {
-			statusUpdate.CompetingTxs = mergeUnique(statusUpdate.CompetingTxs, foundStatusUpdate.CompetingTxs)
-		}
+	err = p.cacheStore.Set(&CacheStatusUpdateHash, status.Hash.String(), bytes, processStatusUpdatesIntervalDefault)
```

Comment:
I don't think it's a good idea to create a TTL that's equal to UpdateInterval, because there may be some hiccups with metamorph service, in which case we can loose data.

Also, this TTL is doing nothing when we're giving a hash to the method (see my next comment)

I would choose either:
1. A very long const TTL in order not to clear records that for some reason were never queried.
2. No TTL at all and periodical clearing cache (what I think we have currently already implemented with `clearCache` on every update.

Reply:
The cache interface was changed, TTL is used only for methods without hash which are currently not used so I'm not using this default value